src/write.jl

```diff
@@ -117,3 +145,36 @@ function Base.write(io::IO, pomdp::POMDP)
 		end
 	end
 end
+
+function pretty_print(io::IO, pomdp::POMDP)
+	_states       = ordered_states( pomdp)
+	_actions      = ordered_actions(pomdp)
+	_observations = ordered_observations(pomdp)
+
+	println("# ------------------------------------------------------------------------")
+	println("# TRANSITIONS")
+	println("T: * : * : * 0.0")
+	for action=_actions, state=_states, statep=_states
+		T = transition(pomdp, state, action)
+		println(io, "T: $(action) : $(state) : $(statep) : $(pdf(T, statep))")
+	end
+	println(io)
+
+	println("# ------------------------------------------------------------------------")
+	println("# OBSERVATIONS")
+	println("O: * : * : * 0.0")
+	for action=_actions, statep=_states, obs=_observations
+		O = observation(pomdp, action, statep)
+		println(io, "O: $(action) : $(statep) : $(obs) $(pdf(O, obs))")
+	end
+	println(io)
+
+	println("# ------------------------------------------------------------------------")
+	println("# REWARDS")
+	println("R: * : * : * : * 0.0")
+	for action=_actions, state=_states, statep=_states, obs=_observations
+		r = reward(pomdp, state, action, statep, obs)
+		println(io, "R: $(action) : $(state) : $(statep) : $(obs) $(r)")
+	end
+	println(io)
+end
```

Comment:
Ah, I did not realize that this function was in this PR. It should not be here unless it has something to do with the POMDP file specification https://pomdp.org/code/pomdp-file-spec.html

Reply:
If you would like to contribute this, it should go in POMDPTools. However, it should also have a more specific name since there are a lot of ways that one might want to "pretty print" a POMDP. I suggest something like `print_distribution_tables`.

Reply:
Hm. I'm not sure I understand what you mean here.

> It should not be here unless it has something to do with the POMDP file specification

Are you referring to lines 154-155, 163-164, and 172-173? While these aren't in the spec, they are valid comments. Perhaps it's the sections are self-evident because they're grouped by `T, O, R` anyways. (So I'm happy to remove it.)

> If you would like to contribute this, it should go in POMDPTools. ...

Perhaps this is why you marked it resolved, but this format is explicitly part of the POMDP spec. The idea behind this formatting is to avoid:
1. Numeric representations.
2. Matrices that are semi-opaque in their representation (because you have to keep in mental memory, or annotate, what columns and rows map on to).

e.g., I work with some folks that are more familiar with the `.pomdp` representation for POMDPs and have never used `POMDPs.jl` (so it's useful for them to have this kind of cleaned representation in the `.pomdp` format since that's the framing they understand).

Reply:
Yeah, I resolved this because I realized that it is part of the spec! Sorry for the confusion!